mixin: Reintroduce thanos_objstore_bucket_operation_failures_total alert

MalloZup opened the pull request:
## Description:

Metric was removed on PR 2002 accidentally. fix #3005 


* [ ] I added CHANGELOG entry for this change.



## Changes

- reintroduce alert `thanos_objstore_bucket_operation_failures_total` which was deleted by mistake

## Verification

tested with make example-rules-lint

yeya24 commented:
Hi, please add this alert to the jsonnet file https://github.com/thanos-io/thanos/blob/master/mixin/alerts/sidecar.libsonnet. 
The markdown files can be auto-generated I think.

MalloZup commented:
@yeya24 thx. I added the jsonnet and generated the markdown via makefile targets. Let me know if something else is missing.

Btw I removed the runbook.md in case is needed I will readd it

MalloZup commented:
I see unit-test are failing I will take look on this.

yeya24 commented:
> I see unit-test are failing I will take look on this.

That's not related to this pr so all good.

MalloZup commented on pkg/rules/rules_test.go:
```diff
@@ -82,7 +82,7 @@ func testRulesAgainstExamples(t *testing.T, dir string, server rulespb.RulesServ
 			Name:                    "thanos-sidecar",
 			File:                    filepath.Join(dir, "alerts.yaml"),
-			Rules:                   []*rulespb.Rule{someAlert, someAlert},
+			Rules:                   []*rulespb.Rule{someAlert, someAlert, someAlert},
```
@yeya24  I have added this `someAlert` fixtures. I was thinking this is needed here since we are adding an alert to the libsonet.

yeya24 replied:
I am not sure. Maybe @kakkoyun or @metalmatze would like to take a look?

yeya24 replied:
Ah yes. Sorry for the late reply. I just checked it again and I think this change is needed. Thanks!

MalloZup replied:
ok thx!

yeya24 commented:
CC @kakkoyun or @metalmatze for a review ^^

kakkoyun commented on mixin/alerts/sidecar.libsonnet:
```diff
@@ -22,6 +22,20 @@
+              summary: 'Thanos Sidecar cannot connect to Prometheus',
+            },
+            expr: |||
+              rate(thanos_objstore_bucket_operation_failures_total{name="prometheus"}[5m]) > 0
```
```suggestion
              rate(thanos_objstore_bucket_operation_failures_total{name="prometheus"}[5m]) > 0
```
Selectors should be templated, check out the other alert examples.

MalloZup replied:
right done

kakkoyun replied:
AFAIT from the CI failures you need to run `make examples`

MalloZup replied:
ah thx!

MalloZup replied:
should be done

kakkoyun reviewed (changes requested):
Looking good. Thanks for your contribution. Only one comment.

MalloZup commented:
@yeya24 @kakkoyun thank you for your reviews. I have fixed your kind comments.

Let me know if other things are missing. Otherwise I will look to a next issue to contribute on :dog2: :service_dog:

yeya24 commented on examples/alerts/alerts.yaml:
```diff
@@ -295,6 +295,17 @@ groups:
+      description: Thanos Sidecar {{$labels.job}} {{$labels.pod}} bucket operations
+        are failing
+      runbook_url: https://github.com/thanos-io/thanos/tree/master/mixin/runbook.md#alert-name-thanossidecarbucketoperationsfailed
+      summary: Thanos Sidecar cannot connect to Prometheus
```
@MalloZup Sorry I missed this. Can you update the summary?

kakkoyun reviewed (approved):
LGTM 🚀 Thanks for the contribution.

kakkoyun commented:
@yeya24 Feel free to merge it if it's also looking good to you.

MalloZup commented:
thx guys for patience and help for getting through this.  :sunflower: 
 I have learned the `mixin` concept of prometheus/grafana, which I wasn't quite aware, so I did definitely discovered something new by this PR :poodle:  

And also jsonnet, was not familiar with it. I knew dhall and toml but  not yet jsonnet. :grin:  That is definitely something I enjoy when picking up new issue on unfamiliar environment .. Thx!

yeya24 reviewed (approved):
LGTM
